### ISSUE TITLE
Tizonia only runs YouTube streams at 160k

### ISSUE THREAD

**Issue by thepyguy** [tizonia/tizonia-openmax-il#741]
# Issue

### Description
Tizonia 

### Steps to Reproduce

1. $ tizonia --youtube-audio-mix-search reggae

**Expected behavior:**
Either option in settings or command line to set to use the higher bitrate HD streams instead of always using default 160k stream.
[What you expect to happen]

**Actual behaviour:**
Always uses 160k stream

**Reproduces how often:**
100%

### Versions

Use the information obtained from `tizonia --debug`.
Debug Info:
	    * [Linux@5.3.0-51-generic-#44~18.04.2-Ubuntu SMP Thu Apr 23 14:27:18 UTC 2020]
	    * [Boost 1_65_1]
	    * [TagLib 1.11.1]
	    * [MediaInfoLib - v17.12]

### Installation Method Used

either debs or built from source snap package is too far behind.

### Additional Information

There are also other issues with the YouTube player for example if I do the following:
1. $ tizonia --youtube-audio-search reggae
I get the following error:
[YouTube] (NameError) : name 'query' is not defined

tizonia exiting (OMX_ErrorInsufficientResources).

 [OMX.Aratelia.audio_source.http:port:0]
 [OMX_ErrorInsufficientResources]

**Comment by juanrubio** [comment]
Hi @thepyguy !

Thanks for stopping by and provide feedback. That's very much appreciated!

> [YouTube] (NameError) : name 'query' is not defined

This issue was fixed a while ago (see #709), although it has not been released (version 0.22.0 is coming out most likely this week).

Regarding making the quality of the YouTube audio streams selectable, it's more difficult that it seems because, for any given YouTube video, only a limited number of audio encodings are available.

For example, take this Katy Perry video, the audio stream can be downloaded in the following formats:
```
$ ytdl -s https://www.youtube.com/watch\?v\=NutHKRKBgR0 -t audio                 
Stream Type    Format Quality         Size            
------ ----    ------ -------         ----            
1      audio   webm   [160k]           3 MB           
2      audio   m4a    [128k]           3 MB           
3      audio   webm   [70k]            1 MB           
4      audio   webm   [50k]            1 MB           
```

So webm 160k is actually the one with the highest quality available. Note that WebM is the container format, not the audio codec. The codec here is OPUS, which is one of the best if not *the* best audio codec available nowadays. OPUS is an evolution of MP3/AAC/SILK/CELT. Check out https://en.wikipedia.org/wiki/Opus_(audio_format) for more information.

So for Tizonia, I decided to focus on YouTube WebM/OPUS streams as the idea is to provide the highest quality available.

Having said that, I'd still like to support m4a formats, in order to support YouTube live streams. This is documented in #391

**Comment by thepyguy** [comment]
Yeah I checked it out with youtube-dl myself. I noticed in the source code webm was selected as the format. Is webm compression a lot less lossy than mp3 etc? If I checkout the latest repo the query issue will be fixed?

I just notice that the highest webm availability is 160k which a 160k mp3 is not very good at all. It appears that although the algorithm has less latency and is better for live audio. I am noticing this repo has lots of code in it some code being c but many scripts being in python also. Did you choose to make core aspects of tizonia in C for performance?
